backend/module/eCampCore/src/EntityService/CampCollaborationService.php

```diff
@@ -45,37 +51,55 @@ protected function createEntity($data) {
 
         $authUser = $this->getAuthUser();
         if (!isset($data->userId)) {
+            $data->userId = null;
```

Comment:
The assumption that `userId` can be null or not present in `$data` makes this function a LOT more complex and harder to understand. What is the exact use case again for that, is it for creating invitations which are not assigned to a user yet? Could we split the code path in the beginning, based on whether there is a user or not? Do you see another way to simplify or split this method?

Reply:
Alright with me :+1: